devices/ak2i.cpp

```diff
@@ -34,26 +33,26 @@ class AK2i : Flashcart {
             // I've been trying to get down to the bottom of this delay for a while
             // hopefully soon it will no longer be needed.
             // ioDelay( 16 * 10 );
-            sendCommand(ak2i_cmdWaitFlashBusy, 4, (uint8_t *)&state, 4);
+            m_card->sendCommand(ak2i_cmdWaitFlashBusy, &state, 4, 4);
             logMessage(LOG_DEBUG, "AK2i: waitFlashBusy = 0x%08x", state);
         } while ((state & 1) != 0);
     }
 
     void a2ki_read(uint8_t *outbuf, uint32_t address) {
-        uint8_t cmdbuf[8];
+        uint8_t cmdbuf[8] = {0};
         logMessage(LOG_DEBUG, "AK2i: read(0x%08x)", address);
         memcpy(cmdbuf, ak2i_cmdReadFlash, 8);
         cmdbuf[1] = (address >> 24) & 0xFF;
         cmdbuf[2] = (address >> 16) & 0xFF;
         cmdbuf[3] = (address >>  8) & 0xFF;
         cmdbuf[4] = (address >>  0) & 0xFF;
 
-        sendCommand(cmdbuf, 0x200, outbuf, 2);
+        m_card->sendCommand(cmdbuf, outbuf, 0x200, 2);
         // a2ki_wait_flash_busy();
     }
 
     void a2ki_erase(uint32_t address) {
-        uint8_t cmdbuf[8];
+        uint8_t cmdbuf[8] = {0};
```

Comment:
I did not initialize this because of the memcpy later that is meant to initialize it. I'm fine with initializing to 0, just want to point that out.